Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -464,13 +505,24 @@ public CsharpCompiler(ErrorReporter reporter)
 
                 Contract.Assert(arg.CompileName == dtor.CorrespondingFormals[n - 1].CompileName);
                 wGet.WriteLine(
-                  $"return (({dt.CompileName}_{dtor.EnclosingCtors[n - 1].CompileName}{DtT_TypeArgs})d).{IdName(arg)}; ");
+                  $"return (({dt.CompileName}_{dtor.EnclosingCtors[n - 1].CompileName}{DtT_TypeArgs})d).{IdName(arg)};");
               }
             }
           }
         }
       }
 
+      foreach (var member in dt.Members) {
```

Comment:
I find the method `CompileDatatypeBase` at ~170 lines already too big, Could you extract this block of code into a separate method? Rider/Resharper make this super easy with the extract method refactoring.

Maybe we can also extract the `// destructors` part into a separate method.